core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -383,59 +329,321 @@ func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.A
 
-	if activeCoordinator == (common.Address{}) {
+	if activeCoordinatorAddress == (common.Address{}) {
 		l.lggr.Error("LogPollerWrapper: cannot update activeCoordinator to zero address")
 		return
 	}
 
-	if activeCoordinator == l.activeCoordinator && proposedCoordinator == l.proposedCoordinator {
+	if activeCoordinatorAddress == l.activeCoordinator.address && proposedCoordinatorAddress == l.proposedCoordinator.address {
 		l.lggr.Debug("LogPollerWrapper: no changes to routes")
 		return
 	}
+
+	activeCoordinatorTypeAndVersion, err := l.getCoordinatorTypeAndVersion(activeCoordinatorAddress)
+	if err != nil {
+		return
+	}
+	activeCoordinator := coordinator{address: activeCoordinatorAddress, typeAndVersion: activeCoordinatorTypeAndVersion}
+
+	proposedCoordinatorTypeAndVersion, err := l.getCoordinatorTypeAndVersion(proposedCoordinatorAddress)
+	if err != nil {
+		return
+	}
+	proposedCoordinator := coordinator{address: proposedCoordinatorAddress, typeAndVersion: proposedCoordinatorTypeAndVersion}
+
 	errActive := l.registerFilters(activeCoordinator)
 	errProposed := l.registerFilters(proposedCoordinator)
 	if errActive != nil || errProposed != nil {
 		l.lggr.Errorw("LogPollerWrapper: Failed to register filters", "errorActive", errActive, "errorProposed", errProposed)
 		return
 	}
 
-	l.lggr.Debugw("LogPollerWrapper: new routes", "activeCoordinator", activeCoordinator.Hex(), "proposedCoordinator", proposedCoordinator.Hex())
+	l.lggr.Debugw("LogPollerWrapper: new routes", "activeCoordinator", activeCoordinator.address.Hex(), "proposedCoordinator", proposedCoordinator.address.Hex())
 	l.activeCoordinator = activeCoordinator
 	l.proposedCoordinator = proposedCoordinator
 
 	for _, subscriber := range l.subscribers {
-		err := subscriber.UpdateRoutes(activeCoordinator, proposedCoordinator)
+		err := subscriber.UpdateRoutes(activeCoordinator.address, proposedCoordinator.address)
 		if err != nil {
 			l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "err", err)
 		}
 	}
 }
 
-func filterName(addr common.Address) string {
-	return logpoller.FilterName("FunctionsLogPollerWrapper", addr.String())
+func filterName(addr common.Address, version string) string {
+	return logpoller.FilterName("FunctionsLogPollerWrapper", addr.String(), "-v", version)
 }
 
-func (l *logPollerWrapper) registerFilters(coordinatorAddress common.Address) error {
-	if (coordinatorAddress == common.Address{}) {
+func (l *logPollerWrapper) registerFilters(coordinator coordinator) error {
+	if (coordinator.address == common.Address{}) {
 		return nil
 	}
-	return l.logPoller.RegisterFilter(
-		logpoller.Filter{
-			Name: filterName(coordinatorAddress),
-			EventSigs: []common.Hash{
-				functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic(),
-				functions_coordinator.FunctionsCoordinatorOracleResponse{}.Topic(),
-			},
-			Addresses: []common.Address{coordinatorAddress},
-		})
+
+	if strings.Contains(coordinator.typeAndVersion, FUNCTIONS_COORDINATOR_VERSION_1) {
```

Comment:
checking TypeAndVersion response from the contract its an exact match.

Why not checking:
```suggestion
	if coordinator.typeAndVersion == FUNCTIONS_COORDINATOR_VERSION_1 {
```

Reply:
you can ignore my comment, you already explained to me you want to catch 1.X changes too 👍🏼